README.md

```diff
@@ -1,2 +1,205 @@
+
+Docker [volumes](https://docs.docker.com/storage/volumes/) are used to mount a user's code into the container.
+Since volumes keep the state of the user's local files in sync with the mounted files in the container, users can develop all code on their machine locally, using their own development tools.
+Then, when they're ready to test/use their code, all they need to do is save their changes and go into the container to build the code and use it.
+The container's purpose is to provide an environment for using/testing Ignition, while everything else (mainly development and version control) should be done on the user's local machine as usual.
+
+_Everything in this repository has been tested/verified using a host machine of Ubuntu 20.04 with an Nvidia GPU (driver 455.28) and Docker version 19.03.13._
+
+## Requirements
+
+* [`Docker`](https://www.docker.com/): Used to provide isolated containers for testing/running Ignition.
+    - [installation instructions](https://docs.docker.com/get-docker/)
+* [`vcstool`](https://github.com/dirk-thomas/vcstool): A tool for working with multiple repositories.
+    - [installation instructions](https://github.com/dirk-thomas/vcstool#how-to-install-vcstool)
+* [`nvidia-docker2`](https://github.com/NVIDIA/nvidia-docker) (if you have an Nvidia GPU): Allows for usage of an Nvidia GPU within Docker containers.
+    - [installation instructions](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker)
+* [`rocker`](https://github.com/osrf/rocker): A tool that allows for greater flexibility with Docker.
+    - [installation instructions](https://github.com/osrf/rocker#installation)
+* [`off-your-rocker`](https://github.com/sloretz/off-your-rocker): A set of extensions for `rocker`.
+    - [installation instructions](https://pypi.org/project/off-your-rocker/) (you may need to install with `pip3` instead of `pip`)
+
+_If you don't have an Nvidia GPU, you can use intel integrated graphics instead._
+
+## Setup
+
+First, make sure you have installed everything listed in the [requirements](#requirements) section.
+
+1. Determine the Ignition version you'd like to use.
+For more information about the versions that are available, take a look at the [releases](https://ignitionrobotics.org/docs).
+Once you've decided, set an environment variable, using all lowercase for the version you chose (in this example, we chose [citadel](https://ignitionrobotics.org/docs/citadel)):
+
+```
+$ export IGN_DISTRO=citadel
+```
+
+2. Build a Docker image that contains the Ignition version specified in step 1.
+The image will contain all of the Ignition binary packages for the version you chose.
+Run the [build.bash script](./build.bash), which requires 2 arguments:
```

Comment:
This seems like something a rocker extension could do just as well.

Reply:
You're right :upside_down_face: maybe an extension like `--ignition <version>`. I'll look into trying to get a rocker extension set up that can replace this